modules/ROOT/pages/dataweave-troubleshoot.adoc

```diff
@@ -109,3 +115,74 @@ Trace:
 When no MIME type is set on the payload, the MIME type defaults to
 `application/java`, and the content is handled as a `String` instead of a
 JSON object.
+
+=== Stack Overflow
+
+When a function recurses too deeply, an error like the following one is thrown:
+
+[source,weave, linenums]
+----
+org.mule.runtime.api.el.ExpressionExecutionException: Stack Overflow. Max stack is 256
+----
+
+You can configure the maximum stack size by using the property `com.mulesoft.dw.stacksize`.
+
+== Output Mismatch When Undefined
+
+Unlike transformations, DataWeave expressions do not require you to define an output format because DataWeave can infer the output based on the expression and the variables you use. Occasionally, the inference process results in a mismatch between the inferred type and the expected type. To resolve this issue, you must make the output explicit. Common examples of this situation occur when:
+
+* <<xml_extraction>>
+* <<multipart_handling>>
+* <<manipulating_text>>
+
+[[xml_extraction]]
+=== Extracting Data from XML
+
+When extracting a String, for example, from an XML payload with the expression `payload.order.product.model`, DataWeave  infers an XML output based on the payload format. In such cases, an error similar to the following one occurs:
+
+[source,weave, linenums]
+----
+"Trying to output non-whitespace characters outside main element tree (in prolog or epilog), while writing Xml at ." evaluating expression: "payload.order.product.model".
+----
+
+For such an error, you must make the output format explicit, for example: `output text/plain --- payload.order.product.model`.
+
+[[multipart_handling]]
+=== Handling Multipart Entries
+
+A common inference error occurs with multipart data, which has a very specific structure. Consider a multipart payload and the expression `dw::core::Objects::keySet(payload.parts)`. Without an explicit output format, DataWeave must infer, based on the payload type, that you intend to output multipart content. In this case, an error similar to the following is thrown:
+
+[source,weave, linenums]
+----
+"Expecting type is {
+  preamble?: String,
+  parts: {
+    _*: {
+      headers: Object,
+      content: Any
+    }
+  }
+} but got Array, while writing MultiPart.
+Trace:
+  at main (Unknown)" evaluating expression: "dw::core::Objects::keySet(payload.parts)".
+----
+
+To resolve this issue, you must define an output format, for example: `output application/json --- dw::core::Objects::keySet(payload.parts)`
+
+[[manipulating_text]]
+=== Manipulating Text Data
+
+Text data can be used to create a more complex object. However, if you do not define the output format for the input text data, DataWeave will infer that it needs to use the plain text writer for the output. The expression `payload splitBy ' '`, for example, will fail with an error similar to:
```

Comment:
"can be used" is in the passive voice; do you intend this? Is this something that DataWeave does or that "you" do?

Reply:
```suggestion
You can use text data to create a more complex object. However, if you do not define the output format for the input text data, DataWeave infers that it must use the plain text writer for the output. The expression `payload splitBy ' '`, for example, will fail with an error similar to:
```